src/lib.rs

```diff
@@ -0,0 +1,589 @@
+    Uint8Array::from(self.to_vec(compress))
+  }
+
+  #[napi]
+  /// Serialize a signature to a hex string.
+  pub fn to_hex(&self, compress: Option<bool>) -> String {
+    format!("0x{}", hex::encode(self.to_vec(compress)))
+  }
+
+  fn to_vec(&self, compress: Option<bool>) -> Vec<u8> {
+    if compress.unwrap_or(true) {
+      return self.0.compress().to_vec();
+    }
+    self.0.serialize().to_vec()
+  }
+
+  #[napi]
+  /// Validate a signature with infinity and group check.
+  ///
+  /// If `sig_infcheck` is `false`, the infinity check will be skipped.
+  pub fn sig_validate(&self, sig_infcheck: Option<bool>) -> Result<Undefined> {
+    min_pk::Signature::validate(&self.0, sig_infcheck.unwrap_or(true)).map_err(from_blst_err)
+  }
+}
+
+//// Exposed functions
+
+#[napi]
+/// Aggregate multiple public keys into a single public key.
+///
+/// If `pks_validate` is `true`, the public keys will be infinity and group checked.
+pub fn aggregate_public_keys(
+  pks: Vec<&PublicKey>,
+  pks_validate: Option<bool>,
+) -> Result<PublicKey> {
+  let pks = pks.iter().map(|pk| &pk.0).collect::<Vec<_>>();
+  min_pk::AggregatePublicKey::aggregate(&pks, pks_validate.unwrap_or(false))
+    .map(|pk| PublicKey(pk.to_public_key()))
+    .map_err(from_blst_err)
+}
+
+#[napi]
+/// Aggregate multiple signatures into a single signature.
+///
+/// If `sigs_groupcheck` is `true`, the signatures will be group checked.
+pub fn aggregate_signatures(
+  sigs: Vec<&Signature>,
+  sigs_groupcheck: Option<bool>,
+) -> Result<Signature> {
+  let sigs = sigs.iter().map(|s| &s.0).collect::<Vec<_>>();
+  min_pk::AggregateSignature::aggregate(&sigs, sigs_groupcheck.unwrap_or(false))
+    .map(|sig| Signature(sig.to_signature()))
+    .map_err(from_blst_err)
+}
+
+#[napi]
+/// Aggregate multiple serialized public keys into a single public key.
+///
+/// If `pks_validate` is `true`, the public keys will be infinity and group checked.
+pub fn aggregate_serialized_public_keys(
+  pks: Vec<Uint8Array>,
+  pks_validate: Option<bool>,
+) -> Result<PublicKey> {
+  let pks = pks.iter().map(|pk| pk.as_ref()).collect::<Vec<_>>();
+  min_pk::AggregatePublicKey::aggregate_serialized(&pks, pks_validate.unwrap_or(false))
+    .map(|pk| PublicKey(pk.to_public_key()))
+    .map_err(from_blst_err)
+}
+
+#[napi]
+/// Aggregate multiple serialized signatures into a single signature.
+///
+/// If `sigs_groupcheck` is `true`, the signatures will be group checked.
+pub fn aggregate_serialized_signatures(
+  sigs: Vec<Uint8Array>,
+  sigs_groupcheck: Option<bool>,
+) -> Result<Signature> {
+  let sigs = sigs.iter().map(|s| s.as_ref()).collect::<Vec<_>>();
+  min_pk::AggregateSignature::aggregate_serialized(&sigs, sigs_groupcheck.unwrap_or(false))
+    .map(|sig| Signature(sig.to_signature()))
+    .map_err(from_blst_err)
+}
+
+#[napi]
+/// Aggregate multiple public keys and multiple serialized signatures into a single blinded public key and blinded signature.
```

Comment:
Was referring to https://en.wikipedia.org/wiki/Blinding_(cryptography)
Specifically this part:
>  In this setting, blinding serves to alter the algorithm's input into some unpredictable state.

and the last paragraph.

That page is a little confusing, and I don't see any part of the page that says "blinding is used in ... context to prevent mutated unverified input from ..." specifically describing our context. But I have heard it used in our context.